crates/ethportal-api/src/types/content_value/beacon.rs

```diff
@@ -512,22 +507,33 @@ impl Decode for ForkVersionedHistoricalSummariesWithProof {
         false
     }
 
-    fn from_ssz_bytes(bytes: &[u8]) -> Result<Self, DecodeError> {
-        Self::decode(bytes)
+    fn ssz_fixed_len() -> usize {
+        ssz::BYTES_PER_LENGTH_OFFSET
     }
-}
 
-impl Encode for ForkVersionedHistoricalSummariesWithProof {
-    fn is_ssz_fixed_len() -> bool {
-        false
-    }
+    fn from_ssz_bytes(bytes: &[u8]) -> Result<Self, DecodeError> {
+        let fork_digest_len = <ForkDigest as Decode>::ssz_fixed_len();
+        let Some((fork_digest_bytes, historical_summaries_with_proof_bytes)) =
+            bytes.split_at_checked(fork_digest_len)
+        else {
+            return Err(DecodeError::InvalidByteLength {
+                len: bytes.len(),
+                expected: fork_digest_len,
+            });
+        };
 
-    fn ssz_append(&self, buf: &mut Vec<u8>) {
-        buf.extend_from_slice(&self.encode());
-    }
+        let fork_digest = ForkDigest::from_ssz_bytes(fork_digest_bytes)?;
```

Comment:
```suggestion
        let fork_digest = ForkDigest::try_from(fork_digest_bytes)?;
```

this shouldn't be ssz no?